noir/noir-repo/tooling/nargo/src/ops/foreign_calls.rs

```diff
@@ -126,10 +132,22 @@ struct ResolveForeignCallRequest<F> {
     #[serde(flatten)]
     /// The foreign call which the external RPC server is to provide a response for.
     function_call: ForeignCallWaitInfo<F>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    /// Root path to the program or workspace in execution.
+    root_path: Option<String>,
```

Comment:
Added extra context required for `TXE`, basically metadata on the project/workspace the nargo command is called from. 

@TomAFrench please LMK if this is a problem, the improvements for us are huge. It doesn't feel like a stretch since it's just metadata added to the RPC call that might be useful in other cases.